plugins/networktables/src/test/java/edu/wpi/first/shuffleboard/plugin/networktables/MarkerGeneratorTest.java

```diff
@@ -0,0 +1,109 @@
+package edu.wpi.first.shuffleboard.plugin.networktables;
+
+import edu.wpi.first.shuffleboard.api.sources.recording.Marker;
+import edu.wpi.first.shuffleboard.api.sources.recording.MarkerImportance;
+import edu.wpi.first.shuffleboard.api.sources.recording.Recorder;
+import edu.wpi.first.shuffleboard.api.util.AsyncUtils;
+import edu.wpi.first.shuffleboard.api.util.FxUtils;
+import edu.wpi.first.shuffleboard.plugin.networktables.util.NetworkTableUtils;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.testfx.util.WaitForAsyncUtils.sleep;
+
+public class MarkerGeneratorTest {
+
+  private static final String MARKER_NAME = "MyEvent";
+
+  private NetworkTableInstance ntInstance;
+  private Recorder recorder;
+  private MarkerGenerator generator;
+  private NetworkTableEntry entry;
+
+  @BeforeEach
+  public void setup() {
+    AsyncUtils.setAsyncRunner(Runnable::run);
+    ntInstance = NetworkTableInstance.create();
+    ntInstance.setUpdateRate(0.01);
+    entry = ntInstance.getEntry(MarkerGenerator.EVENT_TABLE_NAME + MARKER_NAME + MarkerGenerator.EVENT_INFO_KEY);
+    recorder = Recorder.createDummyInstance();
+    generator = new MarkerGenerator(ntInstance, recorder);
+  }
+
+  @AfterEach
+  public void tearDown() {
+    NetworkTableUtils.shutdown(ntInstance);
+    AsyncUtils.setAsyncRunner(FxUtils::runOnFxThread);
+  }
+
+  @Test
+  public void testInvalidEntryType() {
+    entry.setBoolean(false);
+    assertNoMarkersAdded();
+  }
+
+  @Test
+  public void testEmptyStringArray() {
+    entry.setStringArray(new String[0]);
+    assertNoMarkersAdded();
+  }
+
+  @Test
+  public void testStringArrayTooLarge() {
+    entry.setStringArray(new String[3]);
+    assertNoMarkersAdded();
+  }
+
+  private void assertNoMarkersAdded() {
+    waitForEntry();
+    assertEquals(0, recorder.getRecording().getMarkers().size(), "No markers should have been added");
+  }
+
+  private void waitForEntry() {
+    recorder.start();
+    generator.start();
+    sleep(50, TimeUnit.MILLISECONDS);
```

Comment:
Wouldn't it be better to use something like this awaiter library instead of arbitrarily sleeping for some length of time?
https://github.com/jhalterman/concurrentunit